runtime/lua/vim/iter.lua

```diff
@@ -51,6 +51,8 @@
 --- In addition to the |vim.iter()| function, the |vim.iter| module provides
 --- convenience functions like |vim.iter.filter()| and |vim.iter.totable()|.
 
+---@class IterMod
```

Comment:
I don't know much about EmmyLua annotations. Why `IterMod` instead of `Iter`?

Reply:
I just picked a name since `local Iter = {}` has `---@class Iter`. It will probably be issues if there are two tables with the same classname, and it's wrong to say that the class `Iter`'s metamethod `__call` returns `Iter`. I am open to other names though